DotNetPlease/Helpers/MSBuildHelper.cs

```diff
@@ -446,7 +451,24 @@ public static void AddChildren(this ProjectElementContainer container, IEnumerab
         public static void LocateMSBuild()
         {
             if (_msBuildLocated) return;
-            MSBuildLocator.RegisterDefaults();
+
+            var sdkList = ProcessHelper.Run("dotnet", "--list-sdks");
+
+            var sdks = Regex.Matches(sdkList, @"([\d\.]+) *\[(.*)\]")
+                .Select(
+                    m => new
+                    {
+                        Version = new Version(m.Groups[1].Value),
+                        Path = Path.Combine(m.Groups[2].Value, m.Groups[1].Value)
+                    })
+                .OrderBy(x => x.Version)
+                .ToList();
+
+            // TODO: Check global.json and use specific SDK?
+            var sdk = sdks.LastOrDefault();
+            if (sdk == null) throw new InvalidOperationException("Could not locate the .NET SDK");
```

Comment:
Should we provide a way to overwrite this? Eg. when it would have failed, to have a parameter/environment variable/etc to read from.

Reply:
This is what dotnet does when you don't have a global.json. It doesn't really have an impact on how please works, the reason I had to set those variables is that without them, Directory.Build.props is not loaded (I guess that's also something that's implemented in imported .targets files). Ideally we would mimic the SDK selection the dotnet CLI does so that the project files are really loaded by please the same way Visual Studio or dotnet would load them.

Reply:
I actually started implementing that, but then I decided that this is not in scope right now. I just want to solve the issue with the missing SDKs. I tested it locally and now I can run please while not having .NET 6 SDK installed.